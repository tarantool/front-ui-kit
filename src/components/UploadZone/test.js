import React from 'react';
import renderer from 'react-test-renderer';

import { UploadZone } from './index';

const handler = jest.fn();

it('UploadZone renders correctly', () => {
  const tree = renderer
    .create(
      <>
        <UploadZone handler={handler} />
        <UploadZone handler={handler} title="Main title" subTitle="Second title" />
      </>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
