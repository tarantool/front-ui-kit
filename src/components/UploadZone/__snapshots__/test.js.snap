// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadZone renders correctly 1`] = `
Array [
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border-width: 1px;
  border-radius: 4px;
  border-color: #d9dadd;
  border-style: dashed;
  background-color: #fafafa;
  -webkit-transition: border 0.24s ease-in-out;
  transition: border 0.24s ease-in-out;
  outline: none;
  cursor: pointer;
}

.emotion-1 {
  width: 100px;
  height: 80px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  width: 14px;
  height: 14px;
  width: 16px;
  height: 16px;
  width: 100px;
  height: 80px;
  margin-bottom: 16px;
}

.emotion-2 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #040b1d;
  -webkit-font-smoothing: antialiased;
}

<div
    className="emotion-0"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <input
      accept=""
      autoComplete="off"
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <svg
      className="emotion-1"
      viewBox="0 0 100 80"
    >
      <use
        xlinkHref="#icon-drag-file"
      />
    </svg>
    <span
      className="emotion-2"
    >
      Click or drag file to this area to upload
    </span>
  </div>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border-width: 1px;
  border-radius: 4px;
  border-color: #d9dadd;
  border-style: dashed;
  background-color: #fafafa;
  -webkit-transition: border 0.24s ease-in-out;
  transition: border 0.24s ease-in-out;
  outline: none;
  cursor: pointer;
}

.emotion-1 {
  width: 100px;
  height: 80px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  width: 14px;
  height: 14px;
  width: 16px;
  height: 16px;
  width: 100px;
  height: 80px;
  margin-bottom: 16px;
}

.emotion-2 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: #040b1d;
  -webkit-font-smoothing: antialiased;
}

.emotion-3 {
  margin: 0;
  font-family: Inter,Arial,sans-serif;
  font-size: 14px;
  line-height: 22px;
  color: #000000;
  -webkit-font-smoothing: antialiased;
  margin-top: 10px;
  color: #5c606c;
  text-align: center;
  white-space: pre-line;
}

.emotion-3 b {
  font-weight: 600;
}

<div
    className="emotion-0"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <input
      accept=""
      autoComplete="off"
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <svg
      className="emotion-1"
      viewBox="0 0 100 80"
    >
      <use
        xlinkHref="#icon-drag-file"
      />
    </svg>
    <span
      className="emotion-2"
    >
      Main title
    </span>
    <span
      className="emotion-3"
    >
      Second title
    </span>
  </div>,
]
`;
